lib/internal/url.js

```diff
@@ -270,13 +271,17 @@ function onParseHostnameComplete(flags, protocol, username, password,
 
 function onParsePortComplete(flags, protocol, username, password,
                              host, port, path, query, fragment) {
-  this[context].port = port;
+  if ((flags & URL_FLAGS_IS_DEFAULT_SCHEME_PORT) !== 0) {
+    this[context].port = null;
```

Comment:
In the case where `URL_FLAGS_IS_DEFAULT_SCHEME_PORT` is set, `port` should already be `null`. Unless I'm missing something there doesn't seem to be a need to have a conditional here.

Reply:
The reason I added the conditional is because the spec explicitly mentions setting the port to null in case of default scheme port.

I can remove it if you think we can just rely on C++ land to send us null`